test/DirTest.php

```diff
@@ -14,17 +14,24 @@ class DirTest extends TestCase
     /**
      * Ensures that the filter follows expected behavior
      */
-    public function testBasic(): void
+    #[DataProvider('defaultSettingsDataProvider')]
+    public function testBasic(string $input, string $expected): void
     {
-        $filter         = new DirFilter();
-        $valuesExpected = [
-            'filename'              => '.',
-            '/path/to/filename'     => '/path/to',
-            '/path/to/filename.ext' => '/path/to',
+        $filter = new DirFilter();
+
+        self::assertSame($expected, $filter($input));
+        self::assertSame($expected, $filter->__invoke($input));
+        self::assertSame($expected, $filter->filter($input));
+    }
+
+    public static function defaultSettingsDataProvider(): array
```

Comment:
For future reference, please annotate data providers with a precise return types:

```suggestion
    /** @return list<array{0: string, 1: string}> */
    public static function defaultSettingsDataProvider(): array
```

This helps document intent, especially when new items are added or when a larger number of arguments are required.